dropbox/dropbox_client.py

```diff
@@ -223,6 +227,14 @@ def __init__(self,
 
         self._timeout = timeout
 
+    def _check_auth_args(self,
+                         oauth2_access_token,
+                         oauth2_refresh_token,
+                         oauth2_access_token_expiration,
+                         app_key,
+                         app_secret):
+        pass
```

Comment:
not yet implemented?

Reply:
The general idea is that I added a function `_check_auth_args` in the private super class `_DropboxTransport` to allow the different types of client class to override what kind of parameters are allowed in the constructor. E.g. user and team clients use the same checks, so the implementations of `_check_auth_args` in `Dropbox` and `DropboxTeam` both call `_check_user_team_auth_args`, whereas `DropboxAppAuth` has its own implementation that allows only app key and app secret